### PR TITLE
Add pagination to the pages picker

### PR DESCRIPTION
**Pull request by bastianallgeier**
## Describe the PR

This PR adds pagination handling to the pages picker to make it usable with larger lists of pages

## Related issues

- #1632

### CODE REVIEW

File: panel/src/components/Dialogs/PagesDialog.vue [hunk 1]
```diff
@@ -21,42 +21,51 @@
...
-              :disabled="!page.hasChildren"
-              :tooltip="$t('open')"
-              icon="angle-right"
-              @click.stop="go(page)"
-            />
-          </template>
-        </k-list-item>
-      </k-list>
+      <template v-if="models.length">
+        <k-list v-if="models.length">
```
Comment (distantnative) [inline panel/src/components/Dialogs/PagesDialog.vue]: If not needed, already in previous line

**Review by afbora — COMMENTED** [review]
> Pagination page 3 is out of bounds, expected 1-2

I guess we have general `pagination` issue and related #2182 

Close dialog when i stay on third page
Open it again dialog when last single item on last page deleted/change status as draft
See error

How about comparing the current page and the total page when opening the dialog?

**Comment by afbora** [comment]
@bastianallgeier This is related to other PR (merged new pagepicker implement) but when each opening the selection dialog, I noticed that `PagePicker::pagesForQuery()` method works twice. (_on pages() and start() method_) Do we need to cache?

**Comment by bastianallgeier** [comment]
@afbora The dialog needs to reset the page when it is being opened. That's an easy fix!

**Comment by bastianallgeier** [comment]
@afbora I just reset the page when the dialog closes and also cached the result of pagesForQuery.

**Review by afbora — APPROVED** [review]
I think the pages field is great now 🎉

**Pull request merged**